languages/python/omni_http/src/omni_http/omni_httpd/__init__.py

```diff
@@ -15,11 +16,14 @@ class HTTPRequest:
     headers: list[HttpHeader]
 
     def wsgi_environ(self) -> dict:
-        environ = {'wsgi.version': (1, 0), 'wsgi.url_scheme': 'http', 'wsgi.input': self.body or b"",
-                   'wsgi.multithread': False,
+        environ = {'wsgi.version': (1, 0), 'wsgi.url_scheme': 'http', 'wsgi.input': io.BytesIO(self.body or b""), 'wsgi.multithread': False,
                    'REQUEST_METHOD': self.method, 'PATH_INFO': self.path, 'QUERY_STRING': self.query_string or ''}
         for header in self.headers:
-            environ[f"HTTP{header['name'].upper()}"] = header['value']
+            environ[f"HTTP_{header['name'].upper().replace('-', '_')}"] = header['value']
+        
+        # set content-length header to enable body stream
+        if not environ.get('CONTENT_LENGTH'): # TODO: how to handle chunked data?
```

Comment:
I'm leaving out Transfer-Encoding header, because I was not able to get that header. Every curl request with this header is just not being passed/captured for some reason. Maybe because the actual data was not chunked, so curl removes the header. This can be fixed in a future PR.